armi/utils/directoryChangers.py

```diff
@@ -260,7 +260,7 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_value, traceback):
         DirectoryChanger.__exit__(self, exc_type, exc_value, traceback)
         try:
-            pathTools.cleanPath(self.destination)
+            pathTools.cleanPath(self.destination, context.MPI_RANK)
```

Comment:
@john-science 
This change seems to be causing me issues. In particular, during _some_ multi group XS generation runs, my code starts to hang after exiting the temporary directory. 

My code was working before this change, and if I instead use the regular `DirectoryChanger` in place of this, then my code continues as intended.

This appears to be very closely related to @mgjarrett 's fixes in #689. But I honestly have no idea what is the reason why the code hangs...